### PR TITLE
Allow per team approval checks (reviewTeams)

### PR DESCRIPTION
**Pull request by oguzeroglu**
This PR integrates teams into merge-me bot.

Teams can be configured via `merge-me.yml`. An example configuration would be:

```yaml
reviewTeams:
   - design
   - dev
```

Given that configuration the bot will figure out based on requested and actual reviewers what the effective list of teams to review is. __This means that it will only require a review from a given team, if it is included via a mentioned reviewer.__

Per team the existing `minApprovals` is taken into account.

### CODE REVIEW

File: lib/core.js [hunk 1]
```diff
@@ -503,6 +523,180 @@ function isCrossOriginPullRequest(pullRequest) {
...
+
+  try {
+
+    const { head } = pullRequest;
+    const { repo } = head;
+    const { owner } = repo;
+
+    const { type, login } = owner;
+
+    if (type === 'Organization') {
```
Comment (nikku) [inline lib/core.js:536]: Why not for users, too?

In that case I guess we'd need to reject per default?

Comment (oguzeroglu) [reply lib/core.js:536]: In my understanding, the users are not able to have teams in Github, therefore they won't be able to benefit from the reviewTeams functionality of merge-me for their personal repositories.

In that case, this method returns null, which causes `getTeamReviewContext` to return null, which causes team-review mode to switch off.

Comment (nikku) [reply lib/core.js:536]: OK, so our behavior is to simply ignore team reviews in this case. 

Makes sense, I guess.

**Comment by nikku** [comment]
Was not able to push/update your branch so I performed a few changes locally and merged it.

Basically squashed your changes (https://github.com/nikku/merge-me/commit/246c52c375b7e5b7786145f8f45b14440ad19961), applied a number of cleanups on top (mostly test fixtures and test boilerplate improvements), added the now newly required `member:read` permission (https://github.com/nikku/merge-me/commit/558e11401d2b72c433e18a9d09c58cdf1f94e51c) and fixed a crucial bug (https://github.com/nikku/merge-me/commit/bdbec74c64afb2ebb573584ae80937f8f3468e7d).

Thanks for your contribution.

I still think the inner workings are kinda messy. Because of that I'll follow up with a refactoring tomorrow.

**Comment by nikku** [comment]
Merged via https://github.com/nikku/merge-me/commit/246c52c375b7e5b7786145f8f45b14440ad19961.

**Pull request closed without merging**